src/test-component.ts

```diff
@@ -1,22 +1,17 @@
 import { IBaseComponent } from "@well-known-components/interfaces"
-import { pushableChannel } from "@well-known-components/pushable-channel"
 import mitt from "mitt"
-import { INatsComponent, Subscription, NatsEvents, NatsMsg } from "./types"
-
-type PushableChannel = {
-  push: (value: NatsMsg, resolve: (err?: any) => void) => void
-}
+import { INatsComponent, Subscription, NatsEvents, NatsMsg, SubscriptionCallback } from "./types"
 
 /**
  * Create a local NATS component, for testing purposes
  * @public
  */
 export async function createLocalNatsComponent(): Promise<INatsComponent & IBaseComponent> {
-  const channels = new Map<string, PushableChannel>()
+  const callbacks = new Map<string, SubscriptionCallback>()
```

Comment:
Maybe we want many subscriptions to the same topic
```suggestion
  const callbacks = new Map<string, Set<SubscriptionCallback>>()
```